pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -487,73 +487,129 @@ func DetermineLatestKubernetesPatchVersion(cloudProfile gardenv1beta1.CloudProfi
+		return false, []string{}, []gardenv1beta1.OfferedVersion{}, nil
+	}
+
+	return true, newerVersionsString, newerVersions, nil
+}
+
+// GetKubernetesVersionsFromCloudProfile returns the Kubernetes Versions from a CloudProfile
+func GetKubernetesVersionsFromCloudProfile(cloudProfile gardenv1beta1.CloudProfile) ([]gardenv1beta1.OfferedVersion, error) {
 	cloudProvider, err := DetermineCloudProviderInProfile(cloudProfile.Spec)
 	if err != nil {
-		return false, []string{}, err
+		return []gardenv1beta1.OfferedVersion{}, err
 	}
 
-	var (
-		versions      = []string{}
-		newerVersions = []string{}
-	)
+	versions := []gardenv1beta1.OfferedVersion{}
 
 	switch cloudProvider {
 	case gardenv1beta1.CloudProviderAWS:
-		for _, version := range cloudProfile.Spec.AWS.Constraints.Kubernetes.Versions {
+		for _, version := range cloudProfile.Spec.AWS.Constraints.Kubernetes.OfferedVersions {
 			versions = append(versions, version)
 		}
 	case gardenv1beta1.CloudProviderAzure:
-		for _, version := range cloudProfile.Spec.Azure.Constraints.Kubernetes.Versions {
+		for _, version := range cloudProfile.Spec.Azure.Constraints.Kubernetes.OfferedVersions {
 			versions = append(versions, version)
 		}
 	case gardenv1beta1.CloudProviderGCP:
-		for _, version := range cloudProfile.Spec.GCP.Constraints.Kubernetes.Versions {
+		for _, version := range cloudProfile.Spec.GCP.Constraints.Kubernetes.OfferedVersions {
 			versions = append(versions, version)
 		}
 	case gardenv1beta1.CloudProviderOpenStack:
-		for _, version := range cloudProfile.Spec.OpenStack.Constraints.Kubernetes.Versions {
+		for _, version := range cloudProfile.Spec.OpenStack.Constraints.Kubernetes.OfferedVersions {
 			versions = append(versions, version)
 		}
 	case gardenv1beta1.CloudProviderAlicloud:
-		for _, version := range cloudProfile.Spec.Alicloud.Constraints.Kubernetes.Versions {
+		for _, version := range cloudProfile.Spec.Alicloud.Constraints.Kubernetes.OfferedVersions {
 			versions = append(versions, version)
 		}
 	case gardenv1beta1.CloudProviderPacket:
-		for _, version := range cloudProfile.Spec.Packet.Constraints.Kubernetes.Versions {
+		for _, version := range cloudProfile.Spec.Packet.Constraints.Kubernetes.OfferedVersions {
 			versions = append(versions, version)
 		}
 	default:
-		return false, []string{}, fmt.Errorf("unknown cloud provider %s", cloudProvider)
+		return []gardenv1beta1.OfferedVersion{}, fmt.Errorf("unknown cloud provider %s", cloudProvider)
 	}
+	return versions, nil
+}
 
-	for _, version := range versions {
-		ok, err := utils.CompareVersions(version, operator, currentVersion)
-		if err != nil {
-			return false, []string{}, err
-		}
-		if version != currentVersion && ok {
-			newerVersions = append(newerVersions, version)
-		}
+// SetKubernetesVersions sets the Kubernetes Versions to the CloudProfile
+func SetKubernetesVersions(profile *gardenv1beta1.CloudProfile, offeredVersions []gardenv1beta1.OfferedVersion, versions []string) error {
```

Comment:
I don't see this function used anywhere, can we remove it?

Reply:
its being used couple times in the integration tests :)